### PR TITLE
add working implementation of 1D and nD potential landscape computation

### LINKED ISSUE #27: Add n-dimensional attractor dynamics computation
Scalable version of `Wang, S., Falcone, R., Richmond, B. et al. Attractor dynamics reflect decision confidence in macaque prefrontal cortex. Nat Neurosci 26, 1970–1980 (2023).`

### PR DESCRIPTION
**Pull request by kushaangupta**
Closes #27

### CODE REVIEW

File: neuro_py/ensemble/geodyn.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    idx_first_nonzero, idx_last_nonzero = \
+        find_terminal_masked_indices(nonzero_mask, axis=0)  # each have shape: time
+    # along axis 0 set all values from start to idx_first_nonzero to nan
+    for t in range(H.shape[1]):
+        potential_pos_t[:idx_first_nonzero[t], t] = np.nan
+        potential_pos_t[idx_last_nonzero[t] + 1:, t] = np.nan
+
+    return potential_pos_t, grad_pos_t_svm, H, latentedges, domainedges
+
+def potential_landscape_nd(X_dyn, projbins, domainbins=None, nanborderempty=True):
```
Comment (ryanharvey1) [inline neuro_py/ensemble/geodyn.py:87]: add static typing

File: neuro_py/ensemble/geodyn.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+            np.asarray((*X_t_flat.T, norm_tpts)).T, delta_t_flat[:, nnrn], statistic='sum',
+            bins=(*[projbins if isinstance(projbins, int) else projbins[idx] for idx in range(nnrns)], nbins_domain)).statistic
+        # average derivative, a.k.a. flow/vector field for dynamics underlying
+        # population activity
+        grad_pos_t_svm = np.divide(grad_pos_t_svm, H, where=H != 0)
+        grad_pos_t_svm[H == 0] = np.nan  # crucial to handle division by zero
+        # spatial integration via nnancumsum treats nan as zero for cumulative sum
+        potential_pos_t = -np.nancumsum(grad_pos_t_svm, axis=nnrn)  # (nnrns times projbins) x domainbins
+
+        # idx_zero_X_t = np.apply_along_axis(
```
Comment (ryanharvey1) [inline neuro_py/ensemble/geodyn.py:160]: remove if not needed anymore

**Review by ryanharvey1 — COMMENTED** [review]
See comments on static typing. 

Add some type of test for potential_landscape and potential_landscape_nd

**Comment by lolaBerkowitz** [comment]
Make sure everything is formatted in Black. For the notebook, the figures look amazing but please add some descriptions/inferences to all sections. What is the purpose, what is the take home etc.

**Pull request merged**